gprofiler/metadata/application_identifiers.py

```diff
@@ -287,6 +293,35 @@ def get_app_id(self, process: Process) -> Optional[str]:
         return None
 
 
+class _JavaSparkApplicationIdentifier(_ApplicationIdentifier):
+    _JAVA_SPARK_EXECUTOR_ARG = "org.apache.spark.executor.CoarseGrainedExecutorBackend"
+    _SPARK_PROPS_FILE = os.path.join("__spark_conf__", "__spark_conf__.properties")
+    _APP_NAME_NOT_FOUND = "SPARK_APP_NAME_NOT_FOUND"
+    _APP_NAME_KEY = "spark.app.name"
+
+    @staticmethod
+    def _is_java_spark_executor(process: Process) -> bool:
+        args = process.cmdline()
+        if not _is_java_bin(args[0]):
+            return False
+        return _JavaSparkApplicationIdentifier._JAVA_SPARK_EXECUTOR_ARG in args
+
+    def get_app_id(self, process: Process) -> Optional[str]:
+        if not _JavaSparkApplicationIdentifier._is_java_spark_executor(process):
+            return None
+        props_path = os.path.join(process.cwd(), _JavaSparkApplicationIdentifier._SPARK_PROPS_FILE)
+        if not os.path.exists(props_path):
+            _logger.warning("Spark props file doesn't exist")
+            return _JavaSparkApplicationIdentifier._APP_NAME_NOT_FOUND
+        with open(props_path) as f:
+            props_text = f.read()
+        props = dict([line.split("=", 1) for line in props_text.splitlines() if not line.startswith("#")])
+        if _JavaSparkApplicationIdentifier._APP_NAME_KEY in props:
+            return f"java_spark_executor: {props[_JavaSparkApplicationIdentifier._APP_NAME_KEY]}"
```

Comment:
Return only the app name, no need to add the prefix.

Reply:
I disagree, all other appids have prefixes, why not here? `java: spark:` works.

Reply:
We already have prefix from java as we concatenate them, so it's not prefixless.
@liorgorb What do you think?

Reply:
I think it's redundant since it matches a very specific class but I don't really care, can add the prefix to adhere to the style.